var/spack/repos/builtin/packages/python/package.py

```diff
@@ -1100,20 +1106,33 @@ def libs(self):
 
     @property
     def headers(self):
-        # Location where pyconfig.h is _supposed_ to be
-        config_h = self.config_vars["config_h_filename"]
-        if os.path.exists(config_h):
-            headers = HeaderList(config_h)
+        # Locations where pyconfig.h could be
+        # This varies by system, especially on macOS where the command line tools are
+        # installed in a very different directory from the system python interpreter.
+        py_version = str(self.version.up_to(2))
+        candidates = list(
```

Comment:
Idk if we need to convert it back to a list here

Reply:
I guess we can omit it -- do you want me to?

Reply:
Yeah, if it isn't needed, might as well remove it to keep the code simpler

Reply:
fixed